### PR TITLE
FE-1334: Add the functionality on new domain pages to verify domain through url

### PR DESCRIPTION
**Pull request by shrirupa**
FE-1334: Add the functionality on new domain pages to verify domain through url

### What Changed

- Added VerifyToken component on Domains list page
- Whenever domain/list/sending?domain={{domain}}&token={{token}}&mailbox={{mailbox}} is loaded, verify action will trigger on page load
      - if the domain is verified then it would redirect to domain details page with a success alert
      - if domain is not verified then it would stay on the list page with unsuccessful alert
      
      
### How To Test

- Load http://localhost:3100/domains/list/sending?domain=unverified.com&mailbox=abuse&token=akaka url, you should see a unsuccessful alert
- From appteam staging send yourself a email from here 
Screen Shot 2021-01-13 at 5 01 50 PM
     - use the token and mailboax in the link and navigate to http://localhost:3100/domains/list/sending?domain={{domain}}&mailbox={{mailbox}}&token={{token}}

### To Do

- [ ] Address feedback

### CODE REVIEW

File: cypress/tests/integration/configuration/domains/listPage.spec.js [hunk 1]
```diff
@@ -64,6 +64,46 @@ describe('The domains list page', () => {
       cy.url().should('include', `${PAGE_URL}/list/sending`);
     });
 
+    it('successfully verifies the sending/bounce domain when url has token, mailbox and domain as query parameters and redirects to domain details page', () => {
```
Comment (nstanard) [inline cypress/tests/integration/configuration/domains/listPage.spec.js:67]: nit: The domain name used here makes for an odd success message. 😅 
image

File: src/pages/sendingDomains/components/VerifyToken.js [hunk 2]
```diff
@@ -2,7 +2,11 @@ import React, { Component } from 'react'; // eslint-disable-line no-unused-vars
 import { connect } from 'react-redux';
```
Comment (nstanard) [inline src/pages/sendingDomains/components/VerifyToken.js:2]: Note: I believe this is only prettier changes.

File: src/reducers/sendingDomains.js [hunk 3]
```diff
@@ -104,7 +104,7 @@ export default (state = initialState, { type, payload, meta }) => {
     case 'VERIFY_TOKEN_PENDING':
       return {
         ...state,
-        verifyTokenLoading: false,
```
Comment (nstanard) [inline src/reducers/sendingDomains.js:107]: How might I test to make sure this won't cause any issues in the old sending/bounce domain page and loading state?

File: src/pages/sendingDomains/components/VerifyToken.js (hunk 2 above)
Comment (shrirupa) [reply src/pages/sendingDomains/components/VerifyToken.js:2]: yep, just prettier changes.

File: src/reducers/sendingDomains.js (hunk 3 above)
Comment (shrirupa) [reply src/reducers/sendingDomains.js:107]: just did a search for it verifyTokenLoading in vscode, we weren't using it anywhere in the old sending domain pages

File: src/pages/domains/ListPage.js [hunk 4]
```diff
@@ -154,6 +157,13 @@ function DomainTabPages() {
       }}
       loading={listPending || isFirstRender}
```
Comment (nstanard) [inline src/pages/domains/ListPage.js:158]: I wonder if we can/should pass verifyTokenLoading into here too (and set this to && instead of || probably?)

**Comment by shrirupa** [comment]
👀  looking at the failing tests

**Review by nstanard — APPROVED** [review]
I was able to verify a domain via email and follow the link back to the app, where it loaded the details page with a success message. LGTM!

**Pull request merged**